modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/XLowerCaseTokenizerFactory.java

```diff
@@ -20,26 +20,20 @@
 package org.elasticsearch.analysis.common;
 
 import org.apache.lucene.analysis.Tokenizer;
-import org.apache.lucene.analysis.core.LowerCaseTokenizer;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.env.Environment;
 import org.elasticsearch.index.IndexSettings;
 import org.elasticsearch.index.analysis.AbstractTokenizerFactory;
-import org.elasticsearch.index.analysis.MultiTermAwareComponent;
 
-public class LowerCaseTokenizerFactory extends AbstractTokenizerFactory implements MultiTermAwareComponent {
+@Deprecated
+public class XLowerCaseTokenizerFactory extends AbstractTokenizerFactory {
 
-    LowerCaseTokenizerFactory(IndexSettings indexSettings, Environment environment, String name, Settings settings) {
+    public XLowerCaseTokenizerFactory(IndexSettings indexSettings, Environment environment, String name, Settings settings) {
         super(indexSettings, settings);
     }
 
     @Override
     public Tokenizer create() {
-        return new LowerCaseTokenizer();
-    }
-
-    @Override
-    public Object getMultiTermComponent() {
-        return this;
+        return new XLowerCaseTokenizer();
```

Comment:
Can we use a LetterTokenizer followed by a LowerCaseFilter like explained in the deprecation javadocs ?
I don't think we need to keep an `XLowerCaseTokenizer`.

Reply:
I'd hoped to do that, but unfortunately the contract of `create` demands that we return a `Tokenizer`, and I don't think there's an easy way of wrapping a `Tokenizer` + `TokenFilter` combination here?

Reply:
As discussed with @romseygeek offline we'll handle the deprecation and removal in a follow up pr.